ADAL/src/ui/ADWebAuthController.m

```diff
@@ -139,6 +144,15 @@ + (void)startWithRequest:(ADRequestParameters *)requestParams
     webviewConfig.presentationType = ADAuthenticationSettings.sharedInstance.webviewPresentationStyle;
 #endif
 
+    if ([context useRefreshTokenForWebview])
+    {
+        [[webviewConfig customHeaders] setObject:@"1" forKey:ADWebAuthIgnoreSSOHeader];
```

Comment:
should this only be set if refresh token is present?

Reply:
The identity provider only accepts the refresh token if it was acquired in a flow that includes x-ms-sso-Ignore-SSO header. https://github.com/AzureAD/azure-activedirectory-library-for-cpp/blob/f8eefdc3a7e5680b5cebd6f924f9b46e31fc7834/src/cpp/src/lib/windows/WebUITokenRequest.cpp is the c++ SDK implementation, it uses this parameter on every request. We only set it if context explicitly enables it since it causes cookies to be ignored.

Reply:
Good point, thanks for clarifying.